modules/fluid_properties/src/userobjects/TabulatedFluidProperties.C

```diff
@@ -100,11 +108,76 @@ TabulatedFluidProperties::initialSetup()
                    " must be present");
     }
 
+    // Check that any property names read from the file are present in the list of possible
+    // properties, and if they are, add them to the list of read properties
+    for (std::size_t i = 0; i < column_names.size(); ++i)
+    {
+      // Only check properties not in _required_columns
+      if (std::find(_required_columns.begin(), _required_columns.end(), column_names[i]) ==
+          _required_columns.end())
+      {
+        if (std::find(_property_columns.begin(), _property_columns.end(), column_names[i]) ==
+            _property_columns.end())
+          mooseError(column_names[i],
+                     " read in ",
+                     _file_name,
+                     " is not one of the properties that TabulatedFluidProperties understands");
```

Comment:
In general, I really like to see the name of the object that triggered the error. It helps to track down the error in the input file.  You probably do not have heaps of fluid properties in yours, but I still think it is a good idea.

You would change this like so, if you decided to change this:
```
mooseError(name(), "your error message");
```

Reply:
And this